tiledbimg/converters/ome_zarr.py

```diff
@@ -1,24 +1,134 @@
+            store=zarr.storage.DirectoryStore(path=self.output_path), overwrite=True
+        )
+        self.tldb_image = tiledb.Group(input_path, "r")
+        # levels is a list of TLDB ARRAYS
+        self._levels = sorted([resolution.uri for resolution in self.tldb_image])
+        self.level_arrays = [lev_uri for lev_uri in self._levels]
+
+    def uri(self, level: int) -> str:
+        return os.path.join(self.input_path, f"{level}")
+
+    @property
+    def level_count(self) -> int:
+        return len(self.tldb_image)
+
+    def level_image(self, level: int) -> np.ndarray:
+        with tiledb.open(self._levels[level]) as L:
+            temp_array = L[:].swapaxes(0, 2)
+            c, y, x = temp_array.shape
+            tczyx_shape = (1, c, 1, y, x)
+            return temp_array.reshape(tczyx_shape)
+
+    def level_metadata(self, level: int) -> Dict[str, Any]:
+        with tiledb.open(self._levels[level]) as L:
+            return dict(pickle.loads(L.meta["pickled_zarrwriter_kwargs"]))
+
+    def metadata(self) -> Dict[str, Any]:
+        return dict(pickle.loads(self.tldb_image.meta["pickled_zarrwriter_kwargs"]))
 
-from .base import ImageConverter, ImageReader
+    def write(
+        self,
+        images: Sequence[np.ndarray],
+        level_metas: Sequence[Dict[str, Any]],
+        image_meta: Dict[str, Any] = {},
+    ) -> None:
+        level_metas_zarray = []
+        for level_meta in level_metas:
+            zarray_meta = level_meta.get("zarray")
+            if zarray_meta is not None:
+                del zarray_meta.get("compressor")["id"]
+                zarray_meta["compressor"] = Blosc.from_config(
+                    zarray_meta.get("compressor")
+                )
+            level_metas_zarray.append(zarray_meta)
+
+        # Write image does not support incremental pyramid write
+
+        write_multiscale(
+            images,
+            self.image_group,
+            scaler=None,
+            axes=["t", "c", "z", "y", "x"],
+            storage_options=level_metas_zarray,
+            metadata=image_meta.get("metadata"),
+        )
 
 
 class OMEZarrReader(ImageReader):
     def __init__(self, input_path: str):
-        self._zarray = zarr.open(input_path)
+        self.image = Reader(ZarrLocation(input_path))
+        resolutions = [
+            img_path for img_path in glob.glob(f"{input_path}/[!OME]*", recursive=True)
+        ]
+        self.res_nodes = [
+            nodes for res in resolutions for nodes in Reader(ZarrLocation(res))()
+        ]
+        self._levels = []
+        for layer in self.res_nodes:
+            self._levels.append(Level(layer))
 
     @property
     def level_count(self) -> int:
-        return len(self._zarray)
+        return len(self._levels)
 
     def level_image(self, level: int) -> np.ndarray:
-        zarray_l0 = self._zarray[level][0]
-        return np.asarray(zarray_l0).squeeze()
+        leveled_zarray = self._levels[level].node.data[0]
```

Comment:
We should add an assertion for `len(data) == 1` if we expect this to always be the case.